exporters/trace/zipkin/model.go

```diff
@@ -54,8 +56,13 @@ func getServiceName(attrs []attribute.KeyValue) string {
 		}
 	}
 
-	// Resource holds a default value so this might not be reach.
-	return ""
+	// fetch service.name from default resource for backup
+	var defaultServiceName string
+	defaultResource := resource.Default()
+	if value, exists := defaultResource.Set().Value(semconv.ServiceNameKey); exists {
+		defaultServiceName = value.AsString()
+	}
```

Comment:
Can we cache this value after it is determined the first time so this operation isn't needed the second time?

Reply:
Sure, What about move it  to the init function ?